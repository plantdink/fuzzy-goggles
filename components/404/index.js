export * from './Component404';
export { default } from './Component404';
