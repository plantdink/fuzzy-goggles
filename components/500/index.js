export { default } from './Component500';
