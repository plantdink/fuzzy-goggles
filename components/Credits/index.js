export * from './Credits';
export { default } from './Credits';
