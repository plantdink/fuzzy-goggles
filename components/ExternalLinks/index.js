// export { default } from './ExternalLinks';
export * from './ExternalLinks';
