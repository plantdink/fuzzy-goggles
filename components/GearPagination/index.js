export * from './GearPagination';
export { default } from './GearPagination';
