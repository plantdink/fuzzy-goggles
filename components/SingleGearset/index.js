export * from './SingleGearset';
export { default } from './SingleGearset';
