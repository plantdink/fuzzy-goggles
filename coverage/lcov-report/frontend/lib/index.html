
<!doctype html>
<html lang="en">

<head>
    <title>Code coverage report for frontend/lib</title>
    <meta charset="utf-8" />
    <link rel="stylesheet" href="../../prettify.css" />
    <link rel="stylesheet" href="../../base.css" />
    <link rel="shortcut icon" type="image/x-icon" href="../../favicon.png" />
    <meta name="viewport" content="width=device-width, initial-scale=1" />
    <style type='text/css'>
        .coverage-summary .sorter {
            background-image: url(../../sort-arrow-sprite.png);
        }
    </style>
</head>
    
<body>
<div class='wrapper'>
    <div class='pad1'>
        <h1><a href="../../index.html">All files</a> frontend/lib</h1>
        <div class='clearfix'>
            
            <div class='fl pad1y space-right2'>
                <span class="strong">84.61% </span>
                <span class="quiet">Statements</span>
                <span class='fraction'>55/65</span>
            </div>
        
            
            <div class='fl pad1y space-right2'>
                <span class="strong">64% </span>
                <span class="quiet">Branches</span>
                <span class='fraction'>16/25</span>
            </div>
        
            
            <div class='fl pad1y space-right2'>
                <span class="strong">71.42% </span>
                <span class="quiet">Functions</span>
                <span class='fraction'>20/28</span>
            </div>
        
            
            <div class='fl pad1y space-right2'>
                <span class="strong">84.78% </span>
                <span class="quiet">Lines</span>
                <span class='fraction'>39/46</span>
            </div>
        
            
        </div>
        <p class="quiet">
            Press <em>n</em> or <em>j</em> to go to the next uncovered block, <em>b</em>, <em>p</em> or <em>k</em> for the previous block.
        </p>
        <template id="filterTemplate">
            <div class="quiet">
                Filter:
                <input oninput="onInput()" type="search" id="fileSearch">
            </div>
        </template>
    </div>
    <div class='status-line high'></div>
    <div class="pad1">
<table class="coverage-summary">
<thead>
<tr>
   <th data-col="file" data-fmt="html" data-html="true" class="file">File</th>
   <th data-col="pic" data-type="number" data-fmt="html" data-html="true" class="pic"></th>
   <th data-col="statements" data-type="number" data-fmt="pct" class="pct">Statements</th>
   <th data-col="statements_raw" data-type="number" data-fmt="html" class="abs"></th>
   <th data-col="branches" data-type="number" data-fmt="pct" class="pct">Branches</th>
   <th data-col="branches_raw" data-type="number" data-fmt="html" class="abs"></th>
   <th data-col="functions" data-type="number" data-fmt="pct" class="pct">Functions</th>
   <th data-col="functions_raw" data-type="number" data-fmt="html" class="abs"></th>
   <th data-col="lines" data-type="number" data-fmt="pct" class="pct">Lines</th>
   <th data-col="lines_raw" data-type="number" data-fmt="html" class="abs"></th>
</tr>
</thead>
<tbody><tr>
	<td class="file high" data-value="abbreviationsDictionary.js"><a href="abbreviationsDictionary.js.html">abbreviationsDictionary.js</a></td>
	<td data-value="100" class="pic high">
	<div class="chart"><div class="cover-fill cover-full" style="width: 100%"></div><div class="cover-empty" style="width: 0%"></div></div>
	</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="14" class="abs high">14/14</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="12" class="abs high">12/12</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="1" class="abs high">1/1</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="8" class="abs high">8/8</td>
	</tr>

<tr>
	<td class="file high" data-value="displayNumbers.js"><a href="displayNumbers.js.html">displayNumbers.js</a></td>
	<td data-value="90.9" class="pic high">
	<div class="chart"><div class="cover-fill" style="width: 90%"></div><div class="cover-empty" style="width: 10%"></div></div>
	</td>
	<td data-value="90.9" class="pct high">90.9%</td>
	<td data-value="11" class="abs high">10/11</td>
	<td data-value="44.44" class="pct low">44.44%</td>
	<td data-value="9" class="abs low">4/9</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="4" class="abs high">4/4</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="10" class="abs high">10/10</td>
	</tr>

<tr>
	<td class="file high" data-value="displayStrings.js"><a href="displayStrings.js.html">displayStrings.js</a></td>
	<td data-value="84.61" class="pic high">
	<div class="chart"><div class="cover-fill" style="width: 84%"></div><div class="cover-empty" style="width: 16%"></div></div>
	</td>
	<td data-value="84.61" class="pct high">84.61%</td>
	<td data-value="13" class="abs high">11/13</td>
	<td data-value="0" class="pct low">0%</td>
	<td data-value="2" class="abs low">0/2</td>
	<td data-value="85.71" class="pct high">85.71%</td>
	<td data-value="7" class="abs high">6/7</td>
	<td data-value="83.33" class="pct high">83.33%</td>
	<td data-value="12" class="abs high">10/12</td>
	</tr>

<tr>
	<td class="file medium" data-value="testUtils.js"><a href="testUtils.js.html">testUtils.js</a></td>
	<td data-value="74.07" class="pic medium">
	<div class="chart"><div class="cover-fill" style="width: 74%"></div><div class="cover-empty" style="width: 26%"></div></div>
	</td>
	<td data-value="74.07" class="pct medium">74.07%</td>
	<td data-value="27" class="abs medium">20/27</td>
	<td data-value="0" class="pct low">0%</td>
	<td data-value="2" class="abs low">0/2</td>
	<td data-value="56.25" class="pct medium">56.25%</td>
	<td data-value="16" class="abs medium">9/16</td>
	<td data-value="68.75" class="pct medium">68.75%</td>
	<td data-value="16" class="abs medium">11/16</td>
	</tr>

</tbody>
</table>
</div>
                <div class='push'></div><!-- for sticky footer -->
            </div><!-- /wrapper -->
            <div class='footer quiet pad2 space-top1 center small'>
                Code coverage generated by
                <a href="https://istanbul.js.org/" target="_blank" rel="noopener noreferrer">istanbul</a>
                at Tue Apr 05 2022 07:32:49 GMT+1000 (Australian Eastern Standard Time)
            </div>
        <script src="../../prettify.js"></script>
        <script>
            window.onload = function () {
                prettyPrint();
            };
        </script>
        <script src="../../sorter.js"></script>
        <script src="../../block-navigation.js"></script>
    </body>
</html>
    