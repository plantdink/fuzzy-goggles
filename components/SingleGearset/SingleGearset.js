import { gql } from '@apollo/client';
import { stringToParagraphs } from '../../lib/displayStrings';

export const SINGLE_GEARSET_QUERY = gql`
  query SINGLE_GEARSET_QUERY($id: ID!) {
    allGearsets(where: { id: $id }) {
      id
      name
      coreAttribute
      coreAttributeValueLevel40
      coreAttributeValueWT5
      setBonusOne
      setBonusTwo
      setBonusName
      setBonusThree
      setBackpackTalentName
      setBackpackTalent
      setChestTalentName
      setChestTalent
      image {
        id
        altText
        image {
          publicUrlTransformed
        }
      }
      notes
    }
  }
`;

export default function SingleGearset({ gearset }) {
  return (
    <>
      <div className="single-gear-item__content" data-testid="testGearset">
        <div className="single-item__details">
          {stringToParagraphs(gearset.notes)}
          <table>
            <thead>
              <tr className="single-gear-item__title-bar">
                <td className="single-gear-item__subheading">
                  Core Attribute Type - {gearset.coreAttribute}
                </td>
              </tr>
            </thead>
            <tbody>
              <tr>
                <th scope="row">Level 40</th>
                <td>+ {gearset.coreAttributeValueLevel40}</td>
              </tr>
              <tr>
                <th scope="row">World Tier 5</th>
                <td>+ {gearset.coreAttributeValueWT5}</td>
              </tr>
            </tbody>
          </table>

          <table>
            <thead>
              <tr className="single-gear-item__title-bar">
                <td className="single-gear-item__subheading">
                  Gear Set Bonuses
                </td>
              </tr>
            </thead>
            <tbody>
              <tr>
                <td>Whilst wearing:</td>
              </tr>
              <tr>
                <th scope="row" className="single-gear-item__sub-subheading">
                  1 Piece
                </th>
                <td>{gearset.setBonusOne}</td>
              </tr>
              <tr>
                <th scope="row" className="single-gear-item__sub-subheading">
                  2 Piece
                </th>
                <td>{gearset.setBonusTwo}</td>
              </tr>
              <tr>
                <th scope="row" className="single-gear-item__sub-subheading">
                  3 Piece
                </th>
                <td>
                  {gearset.setBonusName}: {gearset.setBonusThree}
                </td>
              </tr>
            </tbody>
          </table>

          <table>
            <thead>
              <tr className="single-gear-item__title-bar">
                <td className="single-gear-item__subheading">
                  Gear Set Talents
                </td>
              </tr>
            </thead>
            <tbody>
              <tr>
                <td>Whilst wearing:</td>
              </tr>
              <tr>
                <th scope="row" className="single-gear-item__sub-subheading">
                  Chest
                </th>
                <td>
                  {gearset.setChestTalentName}: {gearset.setChestTalent}
                </td>
              </tr>
              <tr>
                <th scope="row" className="single-gear-item__sub-subheading">
                  Backpack
                </th>
                <td>
                  {gearset.setBackpackTalentName}: {gearset.setBackpackTalent}
                </td>
              </tr>
            </tbody>
          </table>
        </div>

        <div className="item-image" data-testid="testGearsetImage">
          <img
            className="standard-item"
            id="gearset-page__image"
            src={gearset.image?.image.publicUrlTransformed}
            alt={gearset.image?.altText}
          />
        </div>
      </div>
    </>
  );
}
