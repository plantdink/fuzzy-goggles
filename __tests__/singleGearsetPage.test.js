import { render, screen, cleanup, waitFor } from '@testing-library/react';
import { MockedProvider } from '@apollo/react-testing';
import { fakeGearsetQuery } from '../lib/testUtils';
import { SINGLE_GEARSET_QUERY } from '../components/SingleGearset';
import SingleGearsetPage from '../pages/gear/gearset/[id]';

afterEach(cleanup);

const gearset = fakeGearsetQuery();

const mocks = [
  {
    request: {
      query: SINGLE_GEARSET_QUERY,
      variables: {
        id: gearset.id,
      },
    },
    result: {
      data: {
        allGearsets: [gearset],
      },
    },
  },
];

const errorMocks = [
  {
    request: {
      query: SINGLE_GEARSET_QUERY,
      variables: {
        id: gearset.id,
      },
    },
    result: {
      errors: [
        {
          message: 'Gear Set not found.',
        },
      ],
    },
  },
];

describe('<SingleGearsetPage />', () => {
  it('displays loading when the page first renders', async () => {
    const { container, debug } = render(
      <>
        <MockedProvider mocks={mocks}>
          <SingleGearsetPage query={{ id: `${gearset.id}` }} />
        </MockedProvider>
      </>
    );

    const loaderSpinner = screen.getByTestId('tail-spin-loading');
    expect(loaderSpinner).toBeInTheDocument();
  });

  it('matches the snapshot', async () => {
    const { container, debug } = render(
      <>
        <MockedProvider mocks={mocks}>
          <SingleGearsetPage query={{ id: `${gearset.id}` }} />
        </MockedProvider>
      </>
    );

    await screen.findByTestId('testGearset');
    expect(container).toMatchSnapshot();
  });

  it('loads the data and renders all components with data correctly for a single armour type page', async () => {
    const { container, debug } = render(
      <>
        <MockedProvider mocks={mocks}>
          <SingleGearsetPage query={{ id: `${gearset.id}` }} />
        </MockedProvider>
      </>
    );

    const testTitle = await screen.findByText(gearset.name);
    expect(testTitle).toBeInTheDocument();

    const testSingleGearsetContent = await screen.findByTestId('testGearset');
    expect(testSingleGearsetContent).toBeInTheDocument();
  });

  it('Throws an error when an item is not found', async () => {
    const { container, debug } = render(
      <>
        <MockedProvider mocks={errorMocks}>
          <SingleGearsetPage query={{ id: `${gearset.id}` }} />
        </MockedProvider>
      </>
    );

    expect(screen.queryByTestId('graphql-error')).toBeDefined();
    await waitFor(() => {});
  });
});
