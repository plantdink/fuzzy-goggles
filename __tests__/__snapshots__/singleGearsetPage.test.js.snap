// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SingleGearsetPage /> matches the snapshot 1`] = `
<div>
  <div
    class="SingleArmourPiece__SingleGearItemStyle-sc-1lh23le-0 iZNXSJ"
  >
    <div
      class="single-gear-item__title-bar"
    >
      <h1
        class="single-gear-item__heading"
      >
        Fake Protocol
      </h1>
    </div>
    <div
      class="single-gear-item__content"
      data-testid="testGearset"
    >
      <div
        class="single-item__details"
      >
        <p>
          A Gear Set is a unique combination of equipment pieces that an Agent can obtain.
        </p>
        <p>
           Each Gear Set revolves around a specific operational capability and can strongly influence an agents overall approach to any situation they might face.
        </p>
        <p>
           Unlike a Brand set, the Chest and Backpack from a Gear Set do not have a specific Talent associated with them.
        </p>
        <p>
           Once a certain number of pieces of the same Gear Set are equipped, an agent will obtain various Set bonuses with all the bonuses available when four pieces are equipped.
        </p>
        <p>
           An agent must equip the Chest piece and Backpack in order to access the bonuses associated with each of these items.
        </p>
        <table>
          <thead>
            <tr
              class="single-gear-item__title-bar"
            >
              <td
                class="single-gear-item__subheading"
              >
                Core Attribute Type - 
                Weapon Damage
              </td>
            </tr>
          </thead>
          <tbody>
            <tr>
              <th
                scope="row"
              >
                Level 40
              </th>
              <td>
                + 
                15% Weapon damage
              </td>
            </tr>
            <tr>
              <th
                scope="row"
              >
                World Tier 5
              </th>
              <td>
                + 
                9.9% Weapon Damage
              </td>
            </tr>
          </tbody>
        </table>
        <table>
          <thead>
            <tr
              class="single-gear-item__title-bar"
            >
              <td
                class="single-gear-item__subheading"
              >
                Gear Set Bonuses
              </td>
            </tr>
          </thead>
          <tbody>
            <tr>
              <td>
                Whilst wearing:
              </td>
            </tr>
            <tr>
              <th
                class="single-gear-item__sub-subheading"
                scope="row"
              >
                1 Piece
              </th>
              <td>
                + 15% Marksman Rifle Damage
              </td>
            </tr>
            <tr>
              <th
                class="single-gear-item__sub-subheading"
                scope="row"
              >
                2 Piece
              </th>
              <td>
                + 30% Headshot Damage
              </td>
            </tr>
            <tr>
              <th
                class="single-gear-item__sub-subheading"
                scope="row"
              >
                3 Piece
              </th>
              <td>
                Dead Man's Hand
                : 
                Flip a card when landing shots with a Marksman Rifle. After 5 cards are flipped, the damage of your next shot is amplified by 30%. More shots are enhanced, the better the hand revealed. Four of a Kind: 4 shots. Full House: 3 shots. Aces and Eights: 2 shots. Flip an additional card on headshots.
              </td>
            </tr>
          </tbody>
        </table>
        <table>
          <thead>
            <tr
              class="single-gear-item__title-bar"
            >
              <td
                class="single-gear-item__subheading"
              >
                Gear Set Talents
              </td>
            </tr>
          </thead>
          <tbody>
            <tr>
              <td>
                Whilst wearing:
              </td>
            </tr>
            <tr>
              <th
                class="single-gear-item__sub-subheading"
                scope="row"
              >
                Chest
              </th>
              <td>
                No Limit
                : 
                Increases Dead Man's Hand damage bonus from 30% to 50%.
              </td>
            </tr>
            <tr>
              <th
                class="single-gear-item__sub-subheading"
                scope="row"
              >
                Backpack
              </th>
              <td>
                Ace in the Sleeve
                : 
                Amplifies 1 extra shot when revealing your hand.
              </td>
            </tr>
          </tbody>
        </table>
      </div>
      <div
        class="item-image"
        data-testid="testGearsetImage"
      >
        <img
          alt="Eclipse Protocol gearset icon"
          class="standard-item"
          id="gearset-page__image"
          src="FakeGearsetImage.png"
        />
      </div>
    </div>
  </div>
</div>
`;
